### PR TITLE
Address eslint's warnings

### PR DESCRIPTION
**Pull request by kei95**
Figured making dedicated PR for a fix of eslint as the changes were repetitive and noisy. Haven't done anything special.

### CODE REVIEW

File: src/components/Card/Card.tsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
-import { CSSProperties, ReactChild } from "react";
+import { CSSProperties } from "react";
 import styled from "styled-components";
 
 export interface Props {
   onClick?: () => void;
-  children?: ReactChild | ReactChild[];
+  children?: React.ReactNode;
```
Comment (kei95) [inline src/components/Card/Card.tsx:6]: `ReactChild` is now duplicate and not recommended anymore. Thus I replaced them with `React.ReactNode` which contains all possible JSX's elements as type:
https://github.com/typescript-cheatsheets/react#useful-react-prop-type-examples

File: src/App.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ const GlobalStyle = createGlobalStyle`
   font-family: Open Sans, Montserrat, Roboto
 `;
 
-export default function App() {
+export default function App(): JSX.Element {
```
Comment (kei95) [inline src/App.tsx:7]: I personally like explicitly declaring the functional component's type for a number of reasons. ts-lint was complaining anyway so I figured it'd be a good place to start adding returning type on JSX components

1. It's clear what will you return with the component, sometimes you simply return primitive types such as `string`, `number` etc.
2. When the component has logic to hide itself, you'll need to declare it as `JSX.Element | null` which helps other devs to realize there's a logic to hide the component declaratively

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -17,7 +17,15 @@ export interface Props extends RootBodyProps {
...
 
-export default function Button({ title, icon, ...rootBodyProps }: Props) {
+export default function Button({
+  title,
+  icon,
+  ...rootBodyProps
+}: Props): JSX.Element {
+  if (true) {
+    const [test, setTest] = useState(false);
+  }
```
Comment (kei95) [inline src/components/Button/Button.tsx]: Oops I was messing around with lint😄  Will remove!

File: src/components/Project/Project.stories.tsx [hunk 4]
```diff
@@ -1,14 +1,14 @@
-import styled from "styled-components";
-import Panel from "../Panel";
+// import styled from "styled-components";
+// import Panel from "../Panel";
```
Comment (kei95) [inline src/components/Project/Project.stories.tsx:2]: commented out unused imports for now to suppress baby lints

File: src/components/Button/Button.tsx (hunk 3 above)
Comment (kei95) [reply src/components/Button/Button.tsx]: done

File: src/components/Project/Project.stories.tsx (hunk 4 above)
Comment (ryonryon) [reply src/components/Project/Project.stories.tsx:2]: you can remove them if it is unused

**Review by ryonryon — APPROVED** [review]

**Pull request merged**